packages/node/src/integrations/localvariables.ts

```diff
@@ -0,0 +1,274 @@
+import {
+  ClientOptions,
+  Event,
+  EventProcessor,
+  Exception,
+  Hub,
+  Integration,
+  StackFrame,
+  StackParser,
+} from '@sentry/types';
+import { Debugger, InspectorNotification, Runtime, Session } from 'inspector';
+import { LRUMap } from 'lru_map';
+
+export interface DebugSession {
+  /** Configures and connects to the debug session */
+  configureAndConnect(onPause: (message: InspectorNotification<Debugger.PausedEventDataType>) => void): void;
+  /** Gets local variables for an objectId */
+  getLocalVariables(objectId: string): Promise<Record<string, unknown>>;
+}
+
+/**
+ * Promise API is available as `Experimental` and in Node 19 only.
+ *
+ * Callback-based API is `Stable` since v14 and `Experimental` since v8.
+ * Because of that, we are creating our own `AsyncSession` class.
+ *
+ * https://nodejs.org/docs/latest-v19.x/api/inspector.html#promises-api
+ * https://nodejs.org/docs/latest-v14.x/api/inspector.html
+ */
+class AsyncSession extends Session implements DebugSession {
+  /** @inheritdoc */
+  public configureAndConnect(onPause: (message: InspectorNotification<Debugger.PausedEventDataType>) => void): void {
+    this.connect();
+    this.on('Debugger.paused', onPause);
+    this.post('Debugger.enable');
+    // We only want to pause on uncaught exceptions
+    this.post('Debugger.setPauseOnExceptions', { state: 'uncaught' });
```

Comment:
Does this mean we only add stack locals for uncaught exceptions?

What do you think about making this configurable then?

Reply:
Yes, it's probably worth adding an options object as the first constructor argument so we can also add more options in the future without a breaking change.

Reply:
Let's add that now, and then merge this in? We can keep the options object empty for now.